### ISSUE TITLE
Am I missing some formats?

### ISSUE THREAD

**Issue by tyrosinase** [inkstitch/inkstitch#419]
![2019-03-30--1553983068_1920x1080_scrot](https://user-images.githubusercontent.com/9473609/55282182-ab028b00-5315-11e9-8e4e-b2f3d5363ac6.png)
I only ever see these options to export (either here or in Inkscape>Embroider), and I've seen mentions here of 21 different formats. Am I missing something else that would enable them?

(I am especially excited for HUS because my sister finally found the Reader/Writer device and so I'm gonna be able to run embroidery on the Viking Designer 1+, at least as long as the antique laptop keeps working.)

**Comment by lexelby** [comment]
Sadly, no, you're not missing anything.  We originally did _theoretically_ support 21 formats, through `libembroidery` (part of [EmbroiderModder](https://github.com/Embroidermodder/Embroidermodder)).  However, in reality, most of those output formats didn't actually work properly.

With our rewrite of libembroidery (as [pyembroidery](https://github.com/inkstitch/pyembroidery)), we did fix several output format bugs, resulting in the set of formats in your screenshot.  That's all Ink/Stitch supports for now, but they do at least work properly!  Ink/Stitch can also read several more formats.

Can your sister's machine read any of the more common formats like DST?  Adding HUS support is possible, but it's not easy, especially since I don't have access to a Husqvarna machine.

**Comment by tyrosinase** [comment]
Okay, I just wondered if I was missing a library or something and
everyone else was seeing all these magical formats that I wasn't. I
figured I should be publishing my designs with everything available.
(Jumping spiders for *everyone*!)

And yeah, I'm pretty sure the software that comes with it will be able
to convert things, although the machine itself is ~20 years old. I'll
know more when the stuff arrives (it was Mom's machine; Mom lived with
my sister and some pieces got left behind when I picked it up at
Christmas). Ditto for anyone with a less-common machine who might want
to use my designs.

If and when I get the critter up and running (among other things, it
requires an antique laptop that runs Win95, because newer Windows has an
emulation layer between itself and the parallel port that the DRM device
(the left-behind piece) interprets as a piracy attempt) (Yeah, don't
*even* get me started) I'll be happy to help out with HUS support.

...provided the DRM lets me. :-(

**Comment by lexelby** [comment]
Ugh, what a pain!  If you do take a shot at the HUS format, please let me know if you need any pointers regarding pyembroidery.  The code's a bit quirky but it gets the job done :)

**Comment by wwderw** [comment]
Should be able to emulate Win 95 so not have to keep the antique laptop, if you have a computer that can handle that.  I virtualize Win 98 (slightly different then straight emulation, but I also had to go in a convoluted way since Vbox doesn't support anything before XP out of the box).

**Comment by tyrosinase** [comment]
Nope, it's the virtualization that *causes* the problem. We put a
parallel port card in Mom's desktop machine back in the XP days, and it
doesn't work. 95 lets programs talk directly to the hardware, nothing
newer does that, and that is what the reader/writer (the device that
puts designs on the proprietary cards that actually go in the machine)
insists on.

She could have paid to upgrade the reader/writer to USB before Husqvarna
sold off Viking, and I think those go for about $120 on eBay these days
so I might give that a shot someday if it seems worthwhile. But for now,
I have her old Dell Latitude LM, kept around *just* to drive the
reader/writer, so as long as that hardware holds up the setup should
work. At least, whenever my sister gets around to shipping the missing
box of stuff.

**Comment by tyrosinase** [comment]
Yeah, I'm an old-school Perl coder so that's the biggest obstacle:
Python has just enough similarities to *really* make switching gears
hard. False sense of security and all. :-D

**Comment by wwderw** [comment]
You would pass through that reader hardware to the emulated environment, so the OS could talk directly to it.

That's what I do for an old IR receiver/transmitter that I have.

**Comment by wwderw** [comment]
I'm talking about passing through the reader directly to the VM, so that legacy OS can talk directly to the hardware.

**Comment by tyrosinase** [comment]
Ohhhh, you can *do* that? I guess we thought (assumed? It's been a lot
of years) the VM still virtualized the ports. (Although we went from
"eh, no real need to upgrade the laptop right away anyway" to "well, if
the laptop fails then there's always the $600 USB upgrade" to "well, the
laptop is still going strong and I don't embroider much anymore" and
never really revisited it since she was happy with the setup. So I have
to admit we didn't really chase down all the possibilities.)

Of course, now I run Linux on a machine with no parallel port... at
least the laptop is still going strong. ;-)

**Comment by wwderw** [comment]
You can still get parallel cards now, that you can install after the fact, if your computer can handle that.  Just have to make sure that they have drivers for win 95.  I saw one that had from Win2k up to Win 10, but no Win95 or 98 (I'm sure Win 98 would have the same experience as 95).

**Comment by tyrosinase** [comment]
Yeah, I keep saying "95" but the Latitude is actually Windows 98. (It
does still have the whole set of Win95 install disks, which is probably
where my brain is getting that.)

**Comment by kaalleen** [comment]
@tatarize has worked on this issue:
https://github.com/EmbroidePy/pyembroidery/pull/76

**Comment by mippen** [comment]
> Ugh, what a pain! If you do take a shot at the HUS format, please let me know if you need any pointers regarding pyembroidery. The code's a bit quirky but it gets the job done :)
